x/perpetual/keeper/begin_blocker.go

```diff
@@ -28,46 +28,55 @@ func (k Keeper) BeginBlocker(ctx sdk.Context) {
 			if err != nil {
 				ctx.Logger().Error(err.Error())
 			}
-			err = k.UpdateFundingRate(ctx, &pool)
-			if err != nil {
-				ctx.Logger().Error(err.Error())
-			}
-
-			// account custody from long position
-			totalCustodyLong := sdk.ZeroInt()
-			for _, asset := range pool.PoolAssetsLong {
-				totalCustodyLong = totalCustodyLong.Add(asset.Custody)
-			}
 
-			// account custody from short position
-			totalCustodyShort := sdk.ZeroInt()
-			for _, asset := range pool.PoolAssetsShort {
-				totalCustodyShort = totalCustodyShort.Add(asset.Custody)
-			}
-
-			fundingAmountLong := types.CalcTakeAmount(totalCustodyLong, pool.FundingRate)
-			fundingAmountShort := sdk.ZeroInt()
-
-			fundingRateLong := pool.FundingRate
-			fundingRateShort := sdk.ZeroDec()
+			fundingRateLong, fundingRateShort := k.ComputeFundingRate(ctx, pool)
 
-			// if funding rate is negative, collect from short position
-			if pool.FundingRate.IsNegative() {
-				fundingAmountShort = types.CalcTakeAmount(totalCustodyShort, pool.FundingRate)
-				fundingAmountLong = sdk.ZeroInt()
-
-				fundingRateLong = sdk.ZeroDec()
-				fundingRateShort = pool.FundingRate
-			}
+			// TODO: Change this calculation based on time, going to it in another PR
```

Comment:
@amityadav0 lets make sure we also update the funding payment spec documentation to reflect those changes in the formula, thanks

Reply:
updated funding rate update formula